src/library_webgl.js

```diff
@@ -526,6 +526,21 @@ var LibraryGL = {
       webGLContextAttributes['preserveDrawingBuffer'] = true;
 #endif
 
+#if USE_WEBGL2 && MIN_CHROME_VERSION <= 57
+      // BUG: Workaround Chrome WebGL 2 issue: the first shipped versions of WebGL 2 in Chrome 57 did not actually implement
+      // the new garbage free WebGL 2 entry points that take an offset and a length to an existing heap (instead of having to
+      // create a completely new heap view). In Chrome the entry points only were added in to Chrome 58 and newer. For
+      // Chrome 57 (and older), disable WebGL 2 support altogether.
+      function getChromeVersion() {
+        var chromeVersion = navigator.userAgent.match(/Chrom(e|ium)\/([0-9]+)\./);
+        if (chromeVersion) return chromeVersion[2]|0;
+        // If not chrome, fall through to return undefined. (undefined <= integer will yield false)
+      }
+      if (getChromeVersion() <= 57) {
+        WebGL2RenderingContext = undefined;
```

Comment:
1. How does hiding this prototype definition actually cause Emscripten to disable WebGL 2? (Maybe a comment to explain why it works)
2. That said, I would prefer if we didn't modify the page global state. Is there a way we can disable Emscripten's use of WebGL 2 without doing that?

Reply:
Good points both. Updated to avoid modifying global page state.